dj_database_url.py

```diff
@@ -130,8 +126,6 @@ def parse(
         hostname = spliturl.netloc
         if "@" in hostname:
             hostname = hostname.rsplit("@", 1)[1]
-        if ":" in hostname:
-            hostname = hostname.split(":", 1)[0]
```

Comment:
This came in as part of https://github.com/jazzband/dj-database-url/commit/be4c7f22622d065e454c13cc863f3d4f867cc160 and AFAIK this seems to work without it just fine. We certainly don't have a test case that breaks without it